[PROJECT BRANCH] build: update Spectrum CSS processing to use lightning CSS

Westbrook opened the pull request:
## Description
Base branch for the work to convert Spectrum CSS conversion to the Lightning CSS powered `spectrum-config.v2.js`.

## Types of changes
-   [x] Refactor

## Checklist
-   [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
-   [x] My code follows the code style of this project.
-   [x] If my change required a change to the documentation, I have updated the documentation in this pull request.
-   [x] I have read the **[CONTRIBUTING](<(https://github.com/adobe/spectrum-web-components/blob/main/CONTRIBUTING.md)>)** document.
-   [ ] I have added tests to cover my changes.
-   [x] All new and existing tests passed.

hunterloftis commented:
Checking the potentially-failing VRTs from https://app.circleci.com/pipelines/github/adobe/spectrum-web-components?branch=hloftis%2Ftest-lightningcss-vrts. Starting with HCM:

radio circles disappearing
<img width="454" alt="image" src="https://user-images.githubusercontent.com/364501/226405494-239601b4-736c-49d0-a9b0-255640036a27.png">

virtualized table arrow icon moves down a few pixels
<img width="303" alt="image" src="https://user-images.githubusercontent.com/364501/226405972-af9348ee-1944-43e7-a054-1d73a8451caa.png">

white antialias disappears on switch border (an improvement IMO)
<img width="143" alt="image" src="https://user-images.githubusercontent.com/364501/226406167-492dd89b-c866-49d0-9a99-ac810706b0d8.png">

najikahalsema commented on projects/documentation/content/guides/adding-component.md:
```diff
@@ -48,69 +48,107 @@ must structure our CSS very differently.
+                ),
+                converter.classToId('spectrum-Button-label'),
+                converter.classToSlotted('spectrum-Icon', 'icon'),
                 {
```
Again, some context to what exactly this selector ends up looking like in the css file would be helpful since this is documentation!

najikahalsema replied:
Disregard. I think you cover this in the config documentation!

Westbrook commented:
All of these point to styles that we were previously not rendering correctly.

`sp-radio[readonly]` was displaying the "radio button" even though it _should not_: https://opensource.adobe.com/spectrum-css/radio.html#standard see CSS that was previously not processed at: https://github.com/adobe/spectrum-web-components/blob/912d5f0d0326be21459ae5d13f589d24fb30ea44/packages/radio/src/spectrum-radio.css#L287-L303 due to https://github.com/adobe/spectrum-web-components/blob/912d5f0d0326be21459ae5d13f589d24fb30ea44/packages/radio/src/spectrum-config.v2.js#L93

`sp-table-head-cell[sortable]` is also better aligned to the CSS output: https://opensource.adobe.com/spectrum-css/table.html#sizemedium see previously excluded selector at: https://github.com/adobe/spectrum-web-components/blob/912d5f0d0326be21459ae5d13f589d24fb30ea44/packages/table/src/spectrum-table-head-cell.css#L20-L25 due to being an incorrectly scoped selector

`sp-switch` in WHCM was previously overcorrecting `#input:not(:checked)` as `#input:not([checked])` which is an unattainable state. Reviewing just now, the continues use of `:checked` felt wrong, so we could "normalize" (not sure if it's more normal or less) this as `:host(:not([checked])) #input` with an extra PR if we preferred, visual results should be the same in both cases.

najikahalsema reviewed (commented):
Looks good! Let's see if there are any more vrt changes before merging, but the stuff surrounding it looks pretty good.

hunterloftis commented:
Ok now Express RTL:

Arrow directions have flipped, which to me seems like an improvement:
<img width="592" alt="image" src="https://user-images.githubusercontent.com/364501/226411136-9dde4e1a-ac66-46e6-a94b-dec6e7999d5c.png">

Picker buttons have been offset to the upper-right:
<img width="83" alt="image" src="https://user-images.githubusercontent.com/364501/226411536-85d42506-d2e4-493c-8ad3-52b9a421264e.png">

And like above, radios have lost their circles:
<img width="112" alt="image" src="https://user-images.githubusercontent.com/364501/226411702-a6f7d24a-7eca-432a-a1e1-e31882bee0a8.png">

Also like above, arrow icons have shifted:
<img width="141" alt="image" src="https://user-images.githubusercontent.com/364501/226412589-669c03ad-cffc-4ecc-a644-35a1b2c8c10b.png">

hunterloftis commented:
Classic RTL has no un-mentioned issues.

hunterloftis commented:
I am curious why this failure from one of the included PRs doesn't fail here: https://github.com/adobe/spectrum-web-components/pull/3027#issuecomment-1472214208

Westbrook commented on tasks/process-spectrum.js:
```diff
@@ -0,0 +1,636 @@
+     * @type { import('./spectrum-css-converter').SpectrumCSSConverter}
+     */
+    for await (const conversion of conversions) {
+        // console.log(conversion.name);
```
Oh, yes! I'll add a commit with that clean up.

Westbrook commented:
For Picker Button, the transition comes from actually including the "padding" that is leveraged when composed into patterns like Combobox:

<img width="217" alt="image" src="https://user-images.githubusercontent.com/1156657/226421570-9dfb032e-9190-4c2f-bdbd-5ded0b555470.png">

The space between the border and the button was previously not included.

hunterloftis commented:
Before merging, let's figure out why this diff:

- https://d59266315d1fe98aaf1269eb49d30160--spectrum-web-components.netlify.app/review/#TextfieldStories/quiet.png

Looks different from this diff:
- https://d59266315d1fe98aaf1269eb49d30160--spectrum-web-components.netlify.app/review/#TextfieldStories/quiet.png

Which I would expect to look like this storybook (since the VRTs are based on the stories):
- https://lightningcss--spectrum-web-components.netlify.app/storybook/?path=/story/textfield--quiet

hunterloftis commented:
Those VRTs _are_ reflected in the little test PR here, so we should probably revert & update the hash on this branch: https://f14274fd7ac169c1cc7557693d7e4ad5--spectrum-web-components.netlify.app/review/#TextfieldStories/quiet.png

Westbrook commented:
Things I just did in https://github.com/adobe/spectrum-web-components/pull/3025/commits/214368ea31d93dc57df56bd0daa132f34e9eae3b:
- changed everything to `spectrum-config.js`
- updated the documentation in the types
- simplified the Add a Component docs, so it wasn't as much like the whole config docs
- recycled the VRT hash so can see all the failures in one place, especially https://d59266315d1fe98aaf1269eb49d30160--spectrum-web-components.netlify.app/review/#TextareaStories/quiet.png for @hunterloftis

najikahalsema commented on packages/slider/src/spectrum-config.js:
```diff
@@ -27,7 +27,7 @@ const config = {
             fileName: 'slider',
             hoistCustomPropertiesFrom: 'spectrum-Slider',
-            excludeByExactComponentSeries: [
+            excludeByWholeSelector: [
```
niiiiice :D

najikahalsema reviewed (commented):
Thanks so much for addressing the comments! Waiting in anticipation for the vrts to pass :')

hunterloftis reviewed (approved):
🚢
